### ISSUE TITLE
Readable plural messages in arb files

### ISSUE THREAD

**Issue by shihaohong** [flutter/flutter#47011]
Currently, arb files are parsed as a single massive string as follows:
`"helloWorlds": "{count,plural, =0{Hello}=1{Hello World}=2{Hello two worlds}few{Hello {count} worlds}many{Hello all {count} worlds}other{Hello other {count} worlds}}"`

However, it would be more readable if it could be something like:
```
"helloWorlds": "{
    count,
    plural, 
    =0{Hello}
    =1{Hello World}
    =2{Hello two worlds}
    few{Hello {count} worlds}
    many{Hello all {count} worlds}
    other{Hello other {count} worlds}
}"
```

The problem with this is that newlines are control characters (see [SO post](https://stackoverflow.com/questions/16690101/can-a-json-value-contain-a-multiline-string) for some clarity) so you cannot have a literal newline within the string. 

This means that a workaround would be something like an array of strings:
```
  "helloWorlds": [
    "count",
    "plural", 
    "=0{Hello}",
    "=1{Hello World}"
    "=2{Hello two worlds}",
    "few{Hello {count} worlds}",
    "many{Hello all {count} worlds}",
    "other{Hello other {count} worlds}"
  ]
```

I'd like to hear some thoughts on what our limitations are with how we can define [arb files](https://github.com/google/app-resource-bundle/wiki/ApplicationResourceBundleSpecification#plural-and-gender-support) such that it adheres to the [ICU standard for plural messages](http://userguide.icu-project.org/formatparse/messages). 

I tried concatenating using "+" as in the arb file spec, but using `json.decode` with this formatting resulted in an exception.

cc/ @rami-a @HansMuller

**Comment by HansMuller** [comment]
CC @alan-knight

**Comment by alan-knight** [comment]
That seems tricky. If ARB wants it to be a single string, and JSON doesn't want newlines within a string, I'm not sure there's much to be done. I don't think JSON supports concatenating adjacent strings, which is the only other thing I can think of.

**Comment by HansMuller** [comment]
I suppose if IDEs support editing ARB files and nicely formatting values like this, then the fact that the raw file jams the entire plural expression on one line might be OK.

**Comment by pedromassangocode** [comment]
Is this still a valid issue?

**Comment by shihaohong** [comment]
Yes -- However, there still isn't a clear solution to the problem

**Comment by fzyzcjy** [comment]
Is there any updates?

**Comment by goderbauer** [comment]
@thkim1011 Did any of your recent work enable any of this by any chance?

**Comment by thkim1011** [comment]
No... unfortunately JSON parsing wasn't handled in my recent changes, although I would like to have an alternative extended syntax for .arb files that allows new lines within strings and also allows comments within the .arb file.

**Comment by mosuem** [comment]
I agree with @alan-knight and @HansMuller that this is a fundamental limitation of JSON (see also this nice [SO thread](https://stackoverflow.com/a/52557585)), but luckily, the question is about visual representation. There is a [ARB editor extension](https://github.com/google/arb-editor) for VS Code currently being worked on where this would fit nicely.

**Comment by mosuem** [comment]
After talking with @DanTup: Turns out VS Code at least does not support "virtual" linebreaks, see also https://github.com/microsoft/vscode/issues/63600, so the best we can do is add syntax highlighting for the ICU message format, see also https://github.com/google/arb-editor/pull/1 for a first rough draft.
